### DISCUSSION TITLE
DeepAREstimator Inference takes forever

### DISCUSSION THREAD

**Discussion by DeastinY** [awslabs/gluonts#1352] [General]
Training a DeepAREstimator, then serialize and deserialize it and running `make_evaluation_predictions`, even only with one sample, on new data it's incredibly slow. Stopped it after 30 minutes of trying to create a forecast (through `next(forecast_it)`).
Am I doing something wrong here? Data sizes are small and GPU utilization is constantly at 100% ...
Will try some other approaches and keep this updates.

**Comment by DeastinY** [comment]
I've also tried it with `predictor.predict()` instead. The result is the same (takes forever).
Maybe it's due to around 70ish exogenous dynamic timeseries being included, but it's still inference, right?!
If I use it immediately it doesn't take that long either :/

**Comment by DeastinY** [comment]
Works with `SimpleFeedForwardEstimator` without issues

**Comment by DeastinY** [comment]
in case anyone finds this: NaN in test data was the issue :)

**Reply by mbohlkeschneider** [reply]
That's weird since we handle `NaN` in test data explicitly. If you find time to share the code sample that causes the bug, this would be very useful for us.

**Reply by DeastinY** [reply]
Hi @mbohlkeschneider. It'd be pretty troublesome to decouple from the data, so I'll probably get to it in the near future.
Maybe it's something else and I'm creating the dataset wrong? But adding `df = df.fillna(-1000000)` to mask `NaN` was all it needed to work again :/

**Comment by moghadas76** [comment]
I have the same issue
